spec/services/user_signup_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
+
+describe UserSignup do
+  describe "#sign_up" do
+    context "sending emails" do
+      after {ActionMailer::Base.deliveries.clear}
+      let(:charge) {double(:charge, successful?: true)}
+      before do
+        ActionMailer::Base.deliveries.clear
+        allow(StripeWrapper::Charge).to receive(:create).and_return(charge)
+      end
+      
+      it "sends out email to the user with valid inputs" do
+        UserSignup.new(Fabricate.build(:user, email: "john@example.com", password: "password", name: "john smith")).sign_up("stripetoken", nil)
+        expect(ActionMailer::Base.deliveries.last.to).to eq(["john@example.com"])
+      end
+      
+      it "sends out email containing the user's name with valid inputs" do
+        UserSignup.new(Fabricate.build(:user, email: "john@example.com", password: "password", name: "john smith")).sign_up("stripetoken", nil)
+        expect(ActionMailer::Base.deliveries.last.body).to include("john smith")
+      end
+      
+      it "does not send out email with invalid inputs" do
+        UserSignup.new(User.new(email: "john@example.com")).sign_up("stripetoken", nil)
+        expect(ActionMailer::Base.deliveries).to be_empty
+      end         
+    end      
+    
+    context "with valid personal info and valid card" do
+      it "creates the user" do  
+        charge = double(:charge, successful?: true)
+        expect(StripeWrapper::Charge).to receive(:create).and_return(charge)
+        UserSignup.new(Fabricate.build(:user)).sign_up("stripetoken", nil)
+        expect(User.count).to eq(1)
+      end
+    end
+    
+    context "with valid personal info and declined card" do      
+      it "does not create a new user record" do
+        charge = double(:charge, successful?: false, error_message: "Your card was declined.")
+        expect(StripeWrapper::Charge).to receive(:create).and_return(charge)
+        UserSignup.new(Fabricate.build(:user)).sign_up("stripetoken", nil)            
+        expect(User.count).to eq(0)
+      end
+    end 
+    
+    context "with invalid personal info" do
+      before do
+        UserSignup.new(User.new(name: "Joe", email: "Joesmith@example.com")).sign_up("stripetoken", nil)
+      end
+        
+      it "does not create user" do        
+        expect(User.count).to eq(0)
+      end      
+      
+      it "does not charge the card" do
+        expect(StripeWrapper::Charge).not_to receive(:create)
```

Comment:
The problem in this spec is that you don't assert to anything here. Your `before` action sign up will happen before you mock out `create`. This expectation is not the expectation as all others, this is a mock and must be _before_ action you trigger.

 ```ruby
expect(StripeWrapper::Charge).not_to receive(:create)
UserSignup.new(User.new(name: "Joe", email: "Joesmith@example.com")).sign_up("stripetoken")
```

This way for instance. and the previous `it "does not create user"` can be written differently too.

```ruby
it "does not create user" do
  expect {
    UserSignup.new(User.new(name: "Joe", email: "Joesmith@example.com")).sign_up("stripetoken")
  }.to_not change { User.count }
end
```